lib/web_ui/lib/src/engine/text_editing/text_editing.dart

```diff
@@ -921,13 +967,43 @@ abstract class DefaultTextEditingStrategy implements TextEditingStrategy {
 
     EditingState newEditingState = EditingState.fromDomElement(domElement,
         textCapitalization: _inputConfiguration.textCapitalization);
+    if (composingText != null && newEditingState.text != null) {
+      int composingBase = newEditingState.text!.lastIndexOf(composingText!);
+      if (composingBase >= 0) {
+        newEditingState = newEditingState.copyWith(
+            composingBase: composingBase,
+            composingExtent: composingBase + composingText!.length);
+      }
+    }
```

Comment:
Overall, the PR looks good, except this part. Doing a `lastIndexOf()` doesn't always get you the right index. The cursor could be in the middle of the text when composing starts.

From a quick try, I noticed that `compositionupdate` is always fired before the `input` event. So `composingText` is always up to date by the time we reach the `_handleChange` code here.

Instead of doing a `lastIndexOf()` to find the index of `composingText`, what if we use `newEditingState.baseOffset`? Something like this maybe:
```dart
final int compositionExtent = newEditingState.baseOffset
final int compositionBase = compositionExtent - composingText.length;
```